src/apps/explorer/layout/Header.tsx

```diff
@@ -1,34 +1,70 @@
-import React from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 
-import { Navigation } from 'components/layout/GenericLayout/Navigation'
+import { MenuBarToggle, Navigation } from 'components/layout/GenericLayout/Navigation'
 import { Header as GenericHeader } from 'components/layout/GenericLayout/Header'
 import { NetworkSelector } from 'components/NetworkSelector'
 import { PREFIX_BY_NETWORK_ID, useNetworkId } from 'state/network'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faEllipsisH, faTimes } from '@fortawesome/free-solid-svg-icons'
+import { FlexWrap } from 'apps/explorer/pages/styled'
+import { ExternalLink } from 'components/analytics/ExternalLink'
+import { useHistory } from 'react-router'
 
 export const Header: React.FC = () => {
+  const ref = useRef<HTMLDivElement>(null)
+  const history = useHistory()
+  const [isBarActive, setBarActive] = useState(false)
+
+  useEffect(() => {
+    const isClickedOutside = (e: any): void => {
+      isBarActive && ref.current && !ref.current.contains(e.target) && setBarActive(false)
+    }
+    document.addEventListener('mousedown', isClickedOutside)
+    return (): void => {
+      document.removeEventListener('mousedown', isClickedOutside)
+    }
+  }, [isBarActive])
```

Comment:
Here it would be great to reuse the `useOnClickOutside` hook introduced on [PR686](https://github.com/gnosis/gp-ui/pull/686) and  avoid complaints about using `any`.